api/src/graphql/Post.ts

```diff
@@ -172,5 +171,139 @@ export const PostMutation = extendType({
         });
       },
     });
+
+    t.nonNull.field("deletePost", {
+      type: "Post",
+      args: {
+        id: nonNull(stringArg()),
+      },
+      async resolve(parent, args, context) {
+        const postId = args.id;
+        const { userId } = context;
+
+        if (!userId) {
+          throw new Error("You have to log in");
+        }
+
+        // check if the post exists
+        const postToDelete = (await context.prisma.post.findUnique({
+          where: { id: postId },
+        }));
+        if (!postToDelete) {
+          throw new Error("There is no such post");
+        }
+
+        // return matching point
+        const user = (await context.prisma.user.findUnique({
+          where: { id: userId },
+        })) as User;
+        await context.prisma.user.update({
+          where: { id: userId },
+          data: { matchingPoint: user.matchingPoint + 1 },
+        });
+
+        return context.prisma.post.delete({
+          where: { id: postId }
+        })
+      }
+    });
+
+    t.nonNull.field("updatePost", {
+      type: "Post",
+      args: {
+        id: nonNull(stringArg()),
+        title: stringArg(),
+        description: stringArg(),
+        requiredSkillsIds: list(intArg())
+      },
+      async resolve(parent, args, context) {
+        const { id, title, description, requiredSkillsIds } = args;
+        const { userId } = context;
+        if (!userId) {
+          throw new Error("You have to log in");
+        }
+
+        // check if the post exists
+        const post = await context.prisma.post.findUnique({
+          where: { id: id },
+        });
+        if (!post) {
+          throw new Error("There is no such post");
+        }
+
+        // check if the post is createdBy the user
+        if (post.driverId !== userId) {
+          throw new Error("You do not have rights to update this post");
+        }
+
+        return context.prisma.post.update({
+          where: { id: id },
+          data: {
+            title: title as string,
+            description: description as string,
+            // requiredSkills: {
+            //   connect: requiredSkillsIds.map(id => ({ id })),
+            // }
+          }
+        });
+      }
+    });
+
+    t.nonNull.field("registerNavigator", {
+      type: "Post",
+      args: {
+        postId: nonNull(stringArg()),
+        navigatorId: nonNull(intArg())
+      },
+      async resolve(parent, args, context) {
+        const { postId, navigatorId } = args;
+        const { userId } = context;
+
+        if (!userId) {
+          throw new Error("You have to log in");
+        }
+
+        // check if the post exists
+        const post = await context.prisma.post.findUnique({
+          where: { id: postId },
+        });
+        if (!post) {
+          throw new Error("There is no such post");
+        }
+
+        // check if the post is createdBy the user
+        if (post.driverId !== userId) {
```

Comment:
```suggestion
        if (post.driverId === userId) {
```
ここの比較は「自分自身が投稿したPOSTにマッチできない」だと思うので、
否定ではないかなと思います！